vm/acpi_spec/src/mcfg.rs

```diff
@@ -0,0 +1,154 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+#[cfg(feature = "alloc")]
+pub use self::alloc_parse::*;
+
+use super::Table;
+use crate::packed_nums::*;
+use core::mem::size_of;
+use static_assertions::const_assert_eq;
+use thiserror::Error;
+use zerocopy::FromBytes;
+use zerocopy::Immutable;
+use zerocopy::IntoBytes;
+use zerocopy::KnownLayout;
+use zerocopy::Ref;
+use zerocopy::Unaligned;
+
+#[repr(C)]
+#[derive(Copy, Clone, Debug, IntoBytes, Immutable, KnownLayout, FromBytes, Unaligned)]
+pub struct McfgHeader {
+    pub rsvd: u64_ne,
+}
+
+impl McfgHeader {
+    pub fn new() -> Self {
+        McfgHeader { rsvd: 0.into() }
+    }
+}
+
+impl Table for McfgHeader {
+    const SIGNATURE: [u8; 4] = *b"MCFG";
+}
+
+pub const MCFG_REVISION: u8 = 1;
+
+#[repr(C)]
+#[derive(Copy, Clone, Debug, IntoBytes, Immutable, KnownLayout, FromBytes, Unaligned)]
+pub struct McfgSegmentBusRange {
+    pub ecam_base: u64_ne,
+    pub segment: u16_ne,
+    pub start_bus: u8,
+    pub end_bus: u8,
+    pub rsvd: u32_ne,
+}
```

Comment:
This can be a `bitfield(u128)`, and then you're guaranteed that the size will be correct.

In general, I prefer `bitfield`s for structures that are up to 128 bits in length that need to match some hardware spec. That is perhaps stylistic without any real basis. Interested what you / others think.